src/components/__tests__/InstantSearch.js

```diff
@@ -213,9 +217,44 @@ found in
        <Root>"
 `);
 
-  expect(global.console.warn.mock.calls[0][0]).toMatchInlineSnapshot(
-    `"routing should be an object, with \`router\` and \`stateMapping\`. See https://www.algolia.com/doc/api-reference/widgets/instantsearch/vue/#widget-param-routing"`
-  );
+  expect(warn)
+    .toHaveBeenCalledWith(`The \`routing\` option expects an object with \`router\` and/or \`stateMapping\`.
+
+See https://www.algolia.com/doc/api-reference/widgets/instantsearch/vue/#widget-param-routing`);
+});
+
+it('warns when `routing` does not have `router` or `stateMapping`', () => {
+  mount(InstantSearch, {
+    propsData: {
+      searchClient: {},
+      indexName: 'indexName',
+      routing: {},
+    },
+  });
+
+  expect(warn)
+    .toHaveBeenCalledWith(`The \`routing\` option expects an object with \`router\` and/or \`stateMapping\`.
+
+See https://www.algolia.com/doc/api-reference/widgets/instantsearch/vue/#widget-param-routing`);
+});
+
+it('does not warn when `routing` have either `router` or `stateMapping`', () => {
+  mount(InstantSearch, {
+    propsData: {
+      searchClient: {},
+      indexName: 'indexName',
+      routing: { router: {} },
+    },
+  });
+  mount(InstantSearch, {
+    propsData: {
+      searchClient: {},
+      indexName: 'indexName',
+      routing: { stateMapping: {} },
+    },
+  });
+
+  expect(warn).toHaveBeenCalledTimes(0);
```

Comment:
I find it more clear to use `not.toHaveBeenCalled` here. WDYT?

```suggestion
  expect(warn).not.toHaveBeenCalled();
```

Reply:
We use both in this codebase and mainly `toHaveBeenCalledTimes(0)` in InstantSearch.js because it provides better error messages.